Model/Indexer/AbstractIndexer.php

```diff
@@ -125,18 +125,19 @@ public function __construct(
     /**
      * @param array $ids
      * @suppress PhanTypeMismatchArgument
-     * @throws NostoException
      */
     public function doWork(array $ids = [])
     {
         $userFunctions = [];
 
         switch ($this->getModeSwitcher()->getMode()) {
             case DimensionModeConfiguration::DIMENSION_NONE:
+                /** @var Dimension[] $dimension */
                 foreach ($this->dimensionProvider->getIterator() as $dimension) {
                     if ($this->isDimensionProcessable($dimension)) {
-                        /** @suppress PhanTypeMismatchArgument */
-                        $this->executeByDimensions($dimension, new ArrayIterator($ids));
+                        (function () use ($dimension, $ids) {
```

Comment:
Please sanity check this.

Reply:
It works fine, but I can't suppress the issue since it's in a closure. (Only phan >0.12.x can suppress those)
The thing is that `getIterator` returns a generator that yields an array. But the function itself has a typehint to `\Traversable`